lint-tasks.yml

```diff
@@ -116,20 +105,25 @@ tasks:
           lint-tasks.yml \
           Taskfile.yml
 
-  cpp:
-    internal: true
-    requires:
-      vars: ["FLAGS"]
-    dir: "components/core"
+  cpp-format-check:
+    sources: *cpp_source_files
     deps: ["cpp-lint-configs", "venv"]
     cmds:
-      - |-
-        . "{{.G_LINT_VENV_DIR}}/bin/activate"
-        find src tests \
-          -type f \
-          \( -iname "*.cpp" -o -iname "*.h" -o -iname "*.hpp" -o -iname "*.inc" \) \
-          -print0 | \
-            xargs -0 clang-format {{.FLAGS}} -Werror
+      - task: ":utils:cpp-lint:clang-format"
+        vars:
+          FLAGS: ["--dry-run"]
+          ROOT_PATHS: *cpp_source_files
+          VENV_DIR: "{{.G_LINT_VENV_DIR}}"
+
+  cpp-format-fix:
+    sources: *cpp_source_files
```

Comment:
If we have `sources` here, shall we get rid of the `sources` in `cpp-check`? Same deal with `cpp-fix`.